test/Microsoft.IdentityModel.Protocols.SignedHttpRequest.Tests/SignedHttpRequestValidationTests.cs

```diff
@@ -1683,8 +1683,45 @@ public static TheoryData<ValidateSignedHttpRequestTheoryData> ValidateSignedHttp
                             ValidatedSignedHttpRequest = signedHttpRequest,
                         },
                         TestId = "ValidTest",
+                    },
+                    new ValidateSignedHttpRequestTheoryData
+                    {
+                        SignedHttpRequestToken = signedHttpRequest,
+                        SignedHttpRequestValidationParameters = new SignedHttpRequestValidationParameters()
+                        {
+                            NonceValidatorAsync =  (popKey, signedHttpRequestToken, signedHttpRequestValidationContext, cancellationToken) => false
+                        },
+                        ExpectedException = new ExpectedException(typeof(SignedHttpRequestInvalidNonceClaimException), "IDX23036", typeof(SignedHttpRequestInvalidNonceClaimException)),
+                        TestId = "InValidNonceValidationFailed"
+                    },
+                    new ValidateSignedHttpRequestTheoryData
+                    {
+                        SignedHttpRequestToken = signedHttpRequest,
+                        SignedHttpRequestValidationParameters = new SignedHttpRequestValidationParameters()
+                        {
+                            ValidateB = false,
+                            ValidateH = false,
+                            ValidateM = false,
+                            ValidateP = false,
+                            ValidateQ = false,
+                            ValidateTs = false,
+                            ValidateU = false,
+                            NonceValidatorAsync =  (popKey, signedHttpRequestToken, signedHttpRequestValidationContext, cancellationToken) => true
```

Comment:
This delegate just always returns true, it's nice to have this high level positive case but I think what would be even more helpful and most helpful to Substrate as a sample is showing how you grab the nonce claim off of the signedHttpRequestToken you can set it up to be anything in the test but the body of the delegate should be something like return signedHttpRequestToken.GetClaim... == expectedValueSetupInTest

Reply:
Modify the test case